File: ALODWebUtility/Workflow/WorkStatusOption.cs
using System;
using ALOD.Data;

namespace ALODWebUtility.Workflow
{
    public class WorkStatusOption
    {
        protected bool _active;
        protected int _compo;
        protected byte _groupId;
        protected string _groupName;
        protected int _id;
        protected byte _order;
        protected int _statusOut;
        protected string _statusOutText;
        protected byte _template;
        protected string _text;
        protected bool _valid;
        protected bool _visible;
        protected int _workstatusId;

        public bool Active
        {
            get { return _active; }
            set { _active = value; }
        }

        public int Compo
        {
            get { return _compo; }
            set { _compo = value; }
        }

        public string CompoName
        {
            get
            {
                switch (Compo)
                {
                    case 5:
                        return "ANG";
                    case 6:
                        return "AFRC";
                    case 0:
                        return "All";
                    default:
                        return "UNKNOWN";
                }
            }
        }

        public byte DBSignTemplate
        {
            get { return _template; }
            set { _template = value; }
        }

        public byte GroupId
        {
            get { return _groupId; }
            set { _groupId = value; }
        }

        public string GroupName
        {
            get { return _groupName; }
            set { _groupName = value; }
        }

        public int Id
        {
            get { return _id; }
            set { _id = value; }
        }

        public bool OptionVisible
        {
            get { return _visible; }
            set { _visible = value; }
        }

        public byte SortOrder
        {
            get { return _order; }
            set { _order = value; }
        }

        public int StatusOut
        {
            get { return _statusOut; }
            set { _statusOut = value; }
        }

        public string StatusOutText
        {
            get { return _statusOutText; }
            set { _statusOutText = value; }
        }

        public string Text
        {
            get { return _text; }
            set { _text = value; }
        }

        public bool Valid
        {
            get { return _valid; }
            set { _valid = value; }
        }

        public int WorkStatusId
        {
            get { return _workstatusId; }
            set { _workstatusId = value; }
        }

        public bool Delete()
        {
            SqlDataStore adapter = new SqlDataStore();
            return Convert.ToInt32(adapter.ExecuteNonQuery("core_workstatus_sp_DeleteOption", _id)) > 0;
        }

        public bool Save()
        {
            SqlDataStore adapter = new SqlDataStore();
            _id = Convert.ToInt32(adapter.ExecuteNonQuery("core_workstatus_sp_InsertOption", _id, _workstatusId, _statusOut, _text, _active, _order, _template, _compo));
            return _id > 0;
        }
    }
}

File: ALODWebUtility/Workflow/WorkStatusOption.vb
Imports ALOD.Data

Namespace Worklfow

    Public Class WorkStatusOption

        Protected _active As Boolean

        '07-17-2019 S32019 - Curt Lucas
        Protected _compo As String

        Protected _groupId As Byte
        Protected _groupName As String
        Protected _id As Integer
        Protected _order As Byte
        Protected _statusOut As Integer
        Protected _statusOutText As String
        Protected _template As Byte
        Protected _text As String
        Protected _valid As Boolean
        Protected _visible As Boolean
        Protected _workstatusId As Integer

        Public Property Active() As Boolean
            Get
                Return _active
            End Get
            Set(ByVal value As Boolean)
                _active = value
            End Set
        End Property

        Public Property Compo() As Integer
            Get
                Return _compo
            End Get
            Set(ByVal value As Integer)
                _compo = value
            End Set
        End Property

        Public ReadOnly Property CompoName As String
            Get
                Select Case Compo
                    Case 5
                        Return "ANG"
                    Case 6
                        Return "AFRC"
                    Case 0
                        Return "All"
                    Case Else
                        Return "UNKNOWN"
                End Select
            End Get
        End Property

        Public Property DBSignTemplate() As Byte
            Get
                Return _template
            End Get
            Set(ByVal value As Byte)
                _template = value
            End Set
        End Property

        Public Property GroupId() As Byte
            Get
                Return _groupId
            End Get
            Set(ByVal value As Byte)
                _groupId = value
            End Set
        End Property

        Public Property GroupName() As String
            Get
                Return _groupName
            End Get
            Set(ByVal value As String)
                _groupName = value
            End Set
        End Property

        Public Property Id() As Integer
            Get
                Return _id
            End Get
            Set(ByVal value As Integer)
                _id = value
            End Set
        End Property

        Public Property OptionVisible() As Boolean
            Get
                Return _visible
            End Get
            Set(ByVal value As Boolean)
                _visible = value
            End Set
        End Property

        Public Property SortOrder() As Byte
            Get
                Return _order
            End Get
            Set(ByVal value As Byte)
                _order = value
            End Set
        End Property

        Public Property StatusOut() As Integer
            Get
                Return _statusOut
            End Get
            Set(ByVal value As Integer)
                _statusOut = value
            End Set
        End Property

        Public Property StatusOutText() As String
            Get
                Return _statusOutText
            End Get
            Set(ByVal value As String)
                _statusOutText = value
            End Set
        End Property

        Public Property Text() As String
            Get
                Return _text
            End Get
            Set(ByVal value As String)
                _text = value
            End Set
        End Property

        Public Property Valid() As Boolean
            Get
                Return _valid
            End Get
            Set(ByVal value As Boolean)
                _valid = value
            End Set
        End Property

        Public Property WorkStatusId() As Integer
            Get
                Return _workstatusId
            End Get
            Set(ByVal value As Integer)
                _workstatusId = value
            End Set
        End Property

        Public Function Delete() As Boolean
            Dim adapter As New SqlDataStore
            Return CInt(adapter.ExecuteNonQuery("core_workstatus_sp_DeleteOption", _id)) > 0
        End Function

        Public Function Save() As Boolean
            Dim adapter As New SqlDataStore
            _id = CInt(adapter.ExecuteNonQuery("core_workstatus_sp_InsertOption", _id, _workstatusId, _statusOut, _text, _active, _order, _template, _compo))
            Return _id > 0
        End Function

    End Class

End Namespace